sim/validator.py

```diff
@@ -0,0 +1,146 @@
+    run([
+        "simuG.pl",
+        "-refseq", reference_path,
+        "-snp_count", str(num_snps),
+        "-prefix", output_path + "simulated"
+    ])
+
+
+def simulate_reads(
+    genome_fasta,
+    output_path,
+    read_length=150,
+    seed=1
+):
+    read_1 = output_path + "simulated_S1_R1_X.fastq"
+    read_2 = output_path + "simulated_S1_R2_X.fastq"
+
+    # TODO: store stdout to a file
+    # TODO: set error statistics
+    run([
+        "wgsim",
+        "-1", str(read_length),
+        "-2", str(read_length),
+        "-S", str(seed),
+        "-r", "0",
+        "-R", "0",
+        "-X", "0",
+        "-e", "0",
+        genome_fasta,
+        read_1,
+        read_2
+    ])
+
+    # TODO: use python library instead?
+    run(["gzip", read_1, read_2])
+
+
+def btb_seq(btb_seq_directory, reads_directory, results_directory):
+    run(["bash", "./btb-seq", reads_directory,
+         results_directory], cwd=btb_seq_directory)
+
+
+def performance_test(btb_seq_path, results_path, reference_path, exist_ok=False):
+    """ Runs a performance test against the pipeline
+
+        Parameters:
+            btb_seq_path (str): Path to btb-seq code is stored
+            results_path (str): Output path to performance test results
+            reference_path (str): Path to reference fasta
+            exist_ok (bool): Whether or not to throw an error if a results directory already exists
+
+        Returns:
+            None
+    """
+    # Validate Input
+    if os.path.isdir(results_path):
+        raise Exception("Output results path already exists")
+    os.makedirs(results_path)
+
+    if not os.path.isdir(btb_seq_path):
+        raise Exception("Pipeline code repository not found")
+
+    # Output Directories
+    simulated_genome_path = results_path + 'simulated-genome/'
+    simulated_reads_path = results_path + 'simulated-reads/'
+    btb_seq_backup_path = results_path + 'btb-seq/'
+    btb_seq_results_path = results_path + 'btb-seq-results/'
+
+    # Files generated by the performance pipeline that are passed down
+    # to subsequent processing steps
+    fasta_path = simulated_genome_path + 'simulated.simseq.genome.fa'
+    simulated_snps = simulated_genome_path + "simulated.refseq2simseq.map.txt"
+
+    # Create Output Directories
+    os.makedirs(simulated_genome_path, exist_ok=exist_ok)
+    os.makedirs(simulated_reads_path, exist_ok=exist_ok)
+    os.makedirs(btb_seq_backup_path, exist_ok=exist_ok)
+    os.makedirs(btb_seq_results_path, exist_ok=exist_ok)
+
+    # Backup btb-seq code
+    run(["cp", "-r", btb_seq_path, btb_seq_backup_path])
+
+    # Run Simulation
+    simulate_genome(reference_path, simulated_genome_path)
+    simulate_reads(fasta_path, simulated_reads_path)
+    btb_seq(btb_seq_path, simulated_reads_path, btb_seq_results_path)
+
+    # Analyse Results
+    # HACK: this could easily break if additioanl files are present
+    pipeline_directory = glob.glob(btb_seq_results_path + '*')[0] + '/'
```

Comment:
This broke on my machine: there's a file, "report.html", in btb_seq_results_path which glob was appending to pipeline_directory instead
```suggestion
    pipeline_directory = glob.glob(btb_seq_results_path + 'Results_simulated-reads_*')[0] + '/'
```
works and would be safer.